applications/tests/test_applications_new_schema_ws_queries.py

```diff
@@ -0,0 +1,394 @@
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_order_by_created_at_default(api_client):
+    with freeze_time("2020-02-01"):
+        WinterStorageApplicationFactory()
+
+    with freeze_time("2020-01-01"):
+        WinterStorageApplicationFactory()
+
+    executed = api_client.execute(WINTER_STORAGE_APPLICATIONS_QUERY)
+
+    first_date = isoparse(
+        executed["data"]["winterStorageApplications"]["edges"][0]["node"]["createdAt"]
+    )
+    second_date = isoparse(
+        executed["data"]["winterStorageApplications"]["edges"][1]["node"]["createdAt"]
+    )
+
+    assert first_date < second_date
+
+
+def test_query_winter_storage_application_count(superuser_api_client):
+    count = random.randint(1, 10)
+    for _i in range(count):
+        WinterStorageApplicationFactory()
+
+    query = """
+        {
+            winterStorageApplications {
+                count
+                totalCount
+            }
+        }
+    """
+
+    executed = superuser_api_client.execute(query)
+    assert executed["data"] == {
+        "winterStorageApplications": {"count": count, "totalCount": count}
+    }
+
+
+def test_query_winter_storage_application_count_filtered(
+    superuser_api_client, customer_profile
+):
+    customer_count = random.randint(1, 10)
+    no_customer_count = random.randint(1, 10)
+    total_count = customer_count + no_customer_count
+
+    for _i in range(customer_count):
+        WinterStorageApplicationFactory(customer=customer_profile)
+    for _i in range(no_customer_count):
+        WinterStorageApplicationFactory(customer=None)
+
+    query = """
+        {
+            winterStorageApplications(noCustomer: %s) {
+                count
+                totalCount
+            }
+        }
+    """
+
+    executed = superuser_api_client.execute(query % "true")
+    assert executed["data"] == {
+        "winterStorageApplications": {
+            "count": no_customer_count,
+            "totalCount": total_count,
+        }
+    }
+
+    executed = superuser_api_client.execute(query % "false")
+    assert executed["data"] == {
+        "winterStorageApplications": {
+            "count": customer_count,
+            "totalCount": total_count,
+        }
+    }
+
+
+WINTER_STORAGE_APPLICATION_QUERY = """
+query APPLICATION {
+    winterStorageApplication(id: "%s") {
+        id
+        boatType
+        winterStorageAreaChoices {
+            winterStorageAreaName
+            priority
+        }
+        customer {
+            id
+        }
+    }
+}
+"""
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_application_query(winter_storage_application, api_client):
+    first_choice = WinterAreaChoiceFactory()
+    first_choice.priority = 1
+    first_choice.application = winter_storage_application
+    first_choice.save()
```

Comment:
You can pass those values directly to the factory so you don't have to save it later.
```suggestion
    first_choice = WinterAreaChoiceFactory(priority=1, application=winter_storage_application)
```

Reply:
sweet, I'll change it